tests/test_service.py

```diff
@@ -830,3 +830,183 @@ def testGetAlertWithMultipleTriggers(self):
+            self.assertTrue(isinstance(trigger, Trigger))
+            call_args = tuple(mock_trigger_post.call_args)
+            uri_path = os.path.join(endpoint, "alerts/{}/triggers".format(child_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+    def testAddNotification(self):
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compalert_D), 200)) as mock_add_comp_alert:
+            alert = Alert.from_dict(compalert_D)
+            self.assertTrue(isinstance(alert, Alert))
+            delattr(alert, "id")
+            comp_alert = self.argus.alerts.create_composite_alert(alert)
+            self.assertTrue(isinstance(comp_alert, Alert))
+            call_args = mock_add_comp_alert.call_args
+            uri_path = os.path.join(endpoint, "alerts")
+            self.assertIn((uri_path,), call_args)
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compAlert_notification), 200)) as mock_notification:
+            notification = self.argus.alerts.add_notification_to_composite_alert(comp_alert, Notification.from_dict(compAlert_notification))
+            self.assertTrue(isinstance(notification, Notification))
+            call_args = mock_notification.call_args
+            uri_path = os.path.join(endpoint, "alerts/{}/notifications".format(comp_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+
+    def testDeleteChildAlert(self):
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compalert_D), 200)) as mock_alert:
+            alert = Alert.from_dict(compalert_D)
+            self.assertTrue(isinstance(alert, Alert))
+            delattr(alert, "id")
+            comp_alert = self.argus.alerts.create_composite_alert(alert)
+            self.assertTrue(isinstance(comp_alert, Alert))
+            call_args = mock_alert.call_args
+            uri_path = os.path.join(endpoint, "alerts")
+            self.assertIn((uri_path,), call_args)
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(childAlert_1), 200)):
+            child_alert = self.argus.alerts.add_child_alert_to_composite_alert(comp_alert.id, Alert.from_dict(childAlert_1))
+            self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
+            self.assertTrue(isinstance(child_alert, Alert))
+
+        with mock.patch('requests.Session.delete', return_value=MockResponse("", 200)) as mock_delete:
+            self.argus.alerts.delete_child_alert_from_composite_alert(comp_alert.id, child_alert.id)
+            call_args = mock_delete.call_args
+            uri_path = os.path.join(endpoint, "alerts/{}/children/{}".format(comp_alert.id, child_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+
+    def testDeleteTriggerFromChildAlert(self):
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compalert_D), 200)) as mock_alert:
```

Comment:
Why is the mock.patch part moved into the test?

Reply:
Since i am adding multiple function i thought it would be clean to keep it inside the function

Reply:
Ah ok